src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,198 @@
+"""Large language model (LLM) based policy guided bilevel planning approach.
+
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+from typing import Dict, FrozenSet, Iterator, List, Optional, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.approaches import ApproachFailure
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.settings import CFG
+from predicators.src.structs import Any, GroundAtom, ParameterizedOption, \
+    Sequence, State, _GroundNSRT
+
+
+class LLMProbeApproach(OpenLoopLLMApproach):
+    """LLMProbeApproach definition."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "llm_probe"
+
+    def _get_llm_based_plan(self, state: State, atoms: Set[GroundAtom],
+                     goal: Set[GroundAtom]) -> List[_GroundNSRT]:
+        """In this function, we take predictions the llm outputs regarding 
+        the solution to our PDDL problem and use them to make a dictionary of corresponding 
+        states and actions. This dictionary is used to create a policy which guides 
+        the a star search algorithm in searching for a solution to the PDDL problem."""
+        init = atoms
+        initial_atoms: FrozenSet[GroundAtom] = frozenset(atoms)
+        objects = set(state)
+        # Dictionary used to create policy for search function.
```

Comment:
I think it would be more clear to put the LLM querying helper function inside of this method, and then use the output of that method as input to the second helper function (currently called `get_dictionary`), matching the 3 steps that I described in my previous review